src/include/txn_inline.h

```diff
@@ -115,8 +141,7 @@ __wt_txn_op_set_key(WT_SESSION_IMPL *session, const WT_ITEM *key)
 
     op = txn->mod + txn->mod_count - 1;
 
-    if (WT_SESSION_IS_CHECKPOINT(session) || WT_IS_HS(op->btree->dhandle) ||
```

Comment:
What happened to the WT_SESSION_IS_CHECKPOINT part?

Reply:
It is a duplicate check. As checkpoint can only add changes to metadata. We don't need that.

Reply:
Checkpoint can also add stuff to the history store, but I guess we check for that separately too.